submissions/Roophee/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+class Inhabitant  {
+    constructor(name, gender, saying, legs=4, tail= 1 ){
+        this.name = name;
+        this.gender = gender;
+        this.species = this.constructor.name.toLowerCase();
+        this.saying = saying;
+        this.legs = legs;
+        this.tail = tail;
+        this.friends = [];
+    }
+
+    addFriends(...arrayOfFriends) {
+        arrayOfFriends.forEach(({name}) => this.friends.push(name));
+    }
+
+    description(){
+        let descriptionArray = ['name: '+ this.name,
+            'gender: '+ this.gender,
+            'species: '+ this.species,
+            'saying: '+ this.saying,
+            'legs: '+ this.legs,
+            'tail: '+ this.tail,
+            'friends: '+ (this.friends.length === 0? 'none' : this.friends),
+        ];
+        return descriptionArray.join('; ');
+    }
+}
+
+class Dog extends Inhabitant  {
+    static saying = 'Woof-Woof!';
+    constructor(name, gender){
+        super(name, gender, Dog.saying);
+    }
+}
+
+class Cat extends Inhabitant  {
+    static saying = 'Meow!';
+    constructor(name, gender){
+        super(name, gender, Cat.saying);
+    }
+}
+
+class Human extends Inhabitant  {
+    static saying = 'Cogito ergo sum!';
+    constructor(name, gender,saying, legs= 2, tail = 'none'){
+        super(name, gender, saying || Human.saying, legs, tail)
+        this.hands = 2;
+    }
+    description() {
+        return [super.description(),'hands: '+this.hands].join(';');
```

Comment:
Delimiter was `; ` (with a trailing space) one other time. Worth having a typical delimiter as a constant?

Reply:
Done. Save delimiter in const and use it in join methods.